mountpoint-s3/src/prefetch/parquet_prefetch.rs

```diff
@@ -8,29 +11,88 @@ use mountpoint_s3_client::ObjectClient;
 use parquet::errors::ParquetError;
 use parquet::file::footer::decode_footer;
 use parquet::file::metadata::ParquetMetaData;
+use std::ops::Range;
 use tracing::trace;
 
 use super::PrefetchReadError;
+use crate::checksums::ChecksummedBytes;
+use crate::sync::Arc;
+
+pub use async_lock::RwLock as AsyncRwLock;
 
 const PARQUET_MAGIC_LEN: usize = 8;
 
+/// Range Wrapper for sorting
+/// Ranges are considered equal if both start and end points match exactly
+/// This is used to sort ranges in the interval tree
+/// Primary comparison on start, secondary on end
+#[derive(Debug, Clone)]
+pub struct RangeKey {
+    pub range: Range<u64>,
+}
```

Comment:
This is one of those times that a tuple struct probably is better.

You can instead define it something like:

```rust
pub struct ComparableRange(Range<u64>);
```

I'd rework the Rustdoc to cover why you want to implement this type.

```suggestion
/// Wrapper to allow sorting of [Range<u64>].
///
/// Range does not implement [Ord] as there is no generic meaning,
/// however we need it to implement [Ord] to be used in the interval tree.
/// This type considers ranges to be ordered by first comparing the start of the range, then falling back to the end of the range if equal.
#[derive(Debug, Clone)]
pub struct ComparableRange(pub Range<u64>);
```

(I'd rather we drop that inner `pub`, see https://github.com/PradhayA/mountpoint-s3/pull/3#discussion_r1713914956)

Reply:
Sure this would be great! I'l do this.